src/UserService.Business/EditUserCommand.cs

```diff
@@ -10,27 +10,35 @@ namespace LT.DigitalOffice.UserService.Business
 {
     public class EditUserCommand : IEditUserCommand
     {
-        private readonly IValidator<EditUserRequest> validator;
+        private readonly IValidator<UserRequest> validator;
         private readonly IUserRepository repository;
-        private readonly IMapper<EditUserRequest, DbUser> mapper;
+        private readonly IMapper<UserRequest, DbUser> mapperUser;
+        private readonly IMapper<UserRequest, DbUserCredentials> mapperUserCredentials;
 
         public EditUserCommand(
-            [FromServices] IValidator<EditUserRequest> validator,
+            [FromServices] IValidator<UserRequest> validator,
             [FromServices] IUserRepository repository,
-            [FromServices] IMapper<EditUserRequest, DbUser> mapper)
+            [FromServices] IMapper<UserRequest, DbUser> mapperUser,
+            [FromServices] IMapper<UserRequest, DbUserCredentials> mapperUserCredentials)
         {
             this.validator = validator;
             this.repository = repository;
-            this.mapper = mapper;
+            this.mapperUser = mapperUser;
+            this.mapperUserCredentials = mapperUserCredentials;
         }
 
-        public bool Execute(EditUserRequest request)
+        public bool Execute(UserRequest request)
         {
+            string salt = "Exmple_salt";
+
             validator.ValidateAndThrow(request);
 
-            var user = mapper.Map(request);
+            var dbUser = mapperUser.Map(request);
+            var dbUserCredentials = mapperUserCredentials.Map(request);
+
+            dbUserCredentials.Salt = salt;
 
-            return repository.EditUser(user);
+            return repository.EditUser(dbUser) && repository.EditUserCredentials(dbUserCredentials);
```

Comment:
bad practice, what if something goes wrong in EditUser, should we save Credentials ?

Reply:
it was allowed in messages in slack